### ISSUE TITLE
CDS / CDNSKEY records missing in zone transfers when direct-dnskey=yes

### ISSUE THREAD

**Issue by gryphius** [PowerDNS/pdns#6988]
- Program: Authoritative
 - Issue type: Bug report

### Short description
When `direct-dnskey=yes` is set, CDS and CDNSKEY records are not included in AXFR zone transfers. 

### Environment

 - Operating system: ubuntu 16.04
 - Software version: 
```
root@ns:~# pdns_server --version
Sep 27 05:38:22 PowerDNS Authoritative Server 0.0.2458g628b164 (C) 2001-2018 PowerDNS.COM BV
Sep 27 05:38:22 Using 64-bits mode. Built using gcc 5.4.0 20160609 on Sep 25 2018 13:40:42 by root@16deb14fb81c.
Sep 27 05:38:22 PowerDNS comes with ABSOLUTELY NO WARRANTY. This is free software, and you are welcome to redistribute it according to the terms of the GPL version 2.
Sep 27 05:38:22 Features: sodium openssl lua PKCS#11
Sep 27 05:38:22 Built-in modules:
Sep 27 05:38:22 Configured with: " '--host=' '--build=' '--prefix=/usr' '--sysconfdir=/etc/powerdns' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib' '--with-dynmodules=bind ldap pipe gmysql godbc gpgsql gsqlite3 geoip lua lua2 mydns remote random opendbx tinydns' '--with-modules=' '--enable-tools' '--enable-ixfrdist' '--enable-unit-tests' '--with-lua=luajit' '--enable-experimental-pkcs11' '--enable-systemd' '--with-systemd=/lib/systemd/system' '--enable-libsodium' 'build_alias=' 'host_alias=' 'CFLAGS=-g -O2 -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fstack-protector-strong -Wformat -Werror=format-security'"
```
 - Software source: PowerDNS repository

### Steps to reproduce

1. In the powerdns configuration file, set `direct-dnskey=yes`  and restart the server
2. ```pdnsutil create-zone dnscamel.ch ns.ed25519.ch```
3. ```pdnsutil set-kind dnscamel.ch master```
4. ```pdnsutil secure-zone dnscamel.ch```
5. ```pdnsutil set-publish-cds dnscamel.ch 2```
6. ```pdnsutil set-publish-cdnskey dnscamel.ch```
7. ```dig axfr dnscamel.ch @localhost```

### Expected behaviour

The synthesized CDS/CDNSKEY records are included in the AXFR ( which is the case for `direct-dnskey=no` ) 

```
root@ns:~# dig axfr dnscamel.ch @localhost

; <<>> DiG 9.10.3-P4-Ubuntu <<>> axfr dnscamel.ch @localhost
;; global options: +cmd
dnscamel.ch.            3600    IN      SOA     ns.ed25519.ch. anooky.anooky.ch. 2544 10800 3600 604800 3600
dnscamel.ch.            3600    IN      RRSIG   SOA 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. NLgbB+VY/SoDzWEGExFelSMvcnLaHcNxvFlBh2hl5C0wBjiSTP1PFw5x ObT9NSolTkSCv/qLyY1kDuDFWd0qRA==
dnscamel.ch.            3600    IN      NS      ns.ed25519.ch.
dnscamel.ch.            3600    IN      RRSIG   NS 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. zrsuIO4/O/wUYnNS3huTUE1uxjqbytmuoK/zWrf4eoxnAVJlw9bcVcLq vDGg/l0iTfvilE3mmyK1yO7BDxP0yA==
dnscamel.ch.            3600    IN      CDS     41000 13 2 8FEB053BF16AC31FF1672857206E5A3188A1737B27CA24CC7C547B92 87545F80
dnscamel.ch.            3600    IN      RRSIG   CDS 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. p/ECK9QjUpXJVJtYsO1ncz+1AJfp0l6btCvWaec6AK79G6VL3phB17l1 CkD20Fxz8Q/hJehzQrigAbCtAlLOTg==
dnscamel.ch.            3600    IN      NSEC    dnscamel.ch. NS SOA RRSIG NSEC DNSKEY CDS CDNSKEY
dnscamel.ch.            3600    IN      RRSIG   NSEC 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. vccM6/DxRlnymGrtpVhUvINcHr9ww1wFQrbC7uB+kyYhtRbgAOkPo3o2 rczB1tobW74v0CDfHJ7f0/lMjxHV1g==
dnscamel.ch.            3600    IN      DNSKEY  257 3 13 SKazOyrzl7IiqvI6M+UpakwUUWpxk8un2mdnw/mD4boh0K5Fc3S4PQX9 wK6xlGZR7XO+lcRZyl5rhe81bRqWSA==
dnscamel.ch.            3600    IN      RRSIG   DNSKEY 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. dcMQSS+JsS/4vkauwERvjk7qkfljpTAGAPrlCP+Px0WnKWynvN7chRhM Rrpkk53Lot1x1WdqOfitTHHVRfDyyA==
dnscamel.ch.            3600    IN      CDNSKEY 257 3 13 SKazOyrzl7IiqvI6M+UpakwUUWpxk8un2mdnw/mD4boh0K5Fc3S4PQX9 wK6xlGZR7XO+lcRZyl5rhe81bRqWSA==
dnscamel.ch.            3600    IN      RRSIG   CDNSKEY 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. yKsTe65wBoB0lbfP42X58Y5LFKw8VRuQ6z/ZCVLcX/txS1RMuNcsWieo OsSBJdbytEMMcenuLX01tIofK+sFug==
dnscamel.ch.            3600    IN      SOA     ns.ed25519.ch. anooky.anooky.ch. 2544 10800 3600 604800 3600
;; Query time: 9 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Thu Sep 27 05:35:59 CEST 2018
;; XFR size: 13 records (messages 3, bytes 1152)
```

### Actual behaviour

CDS and CDNSKEY records are not present in the AXFR. 

```
root@ns:~# dig axfr dnscamel.ch @localhost

; <<>> DiG 9.10.3-P4-Ubuntu <<>> axfr dnscamel.ch @localhost
;; global options: +cmd
dnscamel.ch.            3600    IN      SOA     ns.ed25519.ch. anooky.anooky.ch. 2544 10800 3600 604800 3600
dnscamel.ch.            3600    IN      RRSIG   SOA 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. FB+suE0dbPuq4Sfn8LcFHKBm34HuxBo2eR5kzjSc/IbO4kRFYwGxvE7v Sog4Z0JRZYBMHgs/F79F1zRBb1JtEw==
dnscamel.ch.            3600    IN      DNSKEY  257 3 13 SKazOyrzl7IiqvI6M+UpakwUUWpxk8un2mdnw/mD4boh0K5Fc3S4PQX9 wK6xlGZR7XO+lcRZyl5rhe81bRqWSA==
dnscamel.ch.            3600    IN      RRSIG   DNSKEY 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. JUVB0lvFgp7jyP0N2UICkD5plJH1ZMz399EwYM9u9x83xvNrv5Naetxv c2qGhPnPLqa/2vz98NYX2l3H1kjlBw==
dnscamel.ch.            3600    IN      NSEC    dnscamel.ch. NS SOA RRSIG NSEC DNSKEY
dnscamel.ch.            3600    IN      RRSIG   NSEC 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. TxrHYdTvv91h5lvGRszZgExc55iAFPKoubv01fYixyb/01VOzrYjWpvO AOpsU/IaYrhBoy8l6upyEAHq6Q9OJg==
dnscamel.ch.            3600    IN      NS      ns.ed25519.ch.
dnscamel.ch.            3600    IN      RRSIG   NS 13 2 3600 20181011000000 20180920000000 41000 dnscamel.ch. 0vsU+2B5nLaP/qfYosLafXJ6lryCD2EENpU0azm5fpbA//+xhY1FbAqr +jOlR0Fwkp15bkl2U6muPKqj0LNopw==
dnscamel.ch.            3600    IN      SOA     ns.ed25519.ch. anooky.anooky.ch. 2544 10800 3600 604800 3600
;; Query time: 8 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Thu Sep 27 05:34:18 CEST 2018
;; XFR size: 9 records (messages 3, bytes 809)
```

**Comment by pieterlexis** [comment]
> Read additional DNSKEY, CDS and CDNSKEY records from the records table/your BIND zonefile. If not set, DNSKEY, CDS and CDNSKEY records in the zonefiles are ignored.

The docs indeed say "additional"

**Comment by peterthomassen** [comment]
Can this be closed due to #9872, #9873?

**Comment by gryphius** [comment]
> Can this be closed due to #9872, #9873?

Yes, this has been fixed in 4.4.0.  Thanks!
